src/compiler/c.rs

```diff
@@ -660,20 +664,43 @@ pub fn hash_key(
         }
     }
     m.update(preprocessor_output);
+    // clang and clang++ have different behavior despite being byte-for-byte identical binaries, so
+    // we have to incorporate that into the hash as well.
+    m.update(&[plusplus as u8]);
```

Comment:
Let's add this further up near the `compiler_digest` addition` so we don't have to worry about weird interactions with the preprocessor output.

Reply:
Done, let me know if this is in line with what you meant.